streamer/streamer.py

```diff
@@ -439,14 +444,27 @@ def convert_cog(config, output_dir, product, filenames):
         rank = comm.Get_rank()
     except:
         LOG.info("Run with single process")
-        for filename in filenames:
+        if flist is not None:
+            LOG.debug("Open list file %s", flist)
+            with open(flist, 'r') as fb:
+                file_list =  np.genfromtxt(fb, dtype='str')
+        else:
+            file_list = list(filenames)
+        for filename in file_list:
             cog_convert(filename, output_dir)
     else:
         comm.Merge(True)
-        filename = filenames[rank]
-        cog_convert(filename, output_dir)
+        if flist is not None:
+            LOG.debug("Open list file %s", flist)
+            with open(flist, 'r') as fb:
+                file_list =  np.genfromtxt(fb, dtype='str')
+        else:
+            file_list = list(filenames)
```

Comment:
I'm not up on the `MPI` stuff... is it okay to load the file list before calling `comm.Merge(True)` and avoid the duplication?